cpp/oneapi/dal/algo/knn/test/badarg.cpp

```diff
@@ -0,0 +1,232 @@
+}
+
+KNN_BADARG_TEST("throws if neighbor_count is zero in set_neighbor_count") {
+    REQUIRE_THROWS_AS(this->get_descriptor().set_neighbor_count(0), domain_error);
+}
+
+KNN_BADARG_TEST("throws if neighbor_count is negative in constructor") {
+    REQUIRE_THROWS_AS(this->get_descriptor(this->class_count, -1), domain_error);
+}
+
+KNN_BADARG_TEST("throws if neighbor_count is negative in set_neighbor_count") {
+    REQUIRE_THROWS_AS(this->get_descriptor().set_neighbor_count(-1), domain_error);
+}
+
+KNN_BADARG_TEST("throws if class_count = 1 and neighbor_count = 0 in constructor") {
+    REQUIRE_THROWS_AS(this->get_descriptor(0, 0), domain_error);
+}
+
+KNN_BADARG_TEST("throws if both class_count and neighbor_count are zero in constructor") {
+    REQUIRE_THROWS_AS(this->get_descriptor(0, 0), domain_error);
+}
+
+KNN_BADARG_TEST("throws if both class_count and neighbor_count are negative in constructor") {
+    REQUIRE_THROWS_AS(this->get_descriptor(-1, -1), domain_error);
+}
+
+KNN_BADARG_TEST("accepts train data and labels") {
+    CAPTURE(this->not_available_on_device());
+    SKIP_IF(this->not_available_on_device());
+    const auto knn_desc = this->get_descriptor();
+    REQUIRE_NOTHROW(this->train(knn_desc, this->get_train_data(), this->get_train_labels()));
+}
+
+KNN_BADARG_TEST("throws if both train data and labels are empty") {
+    SKIP_IF(this->not_available_on_device());
+    const auto knn_desc = this->get_descriptor();
+    REQUIRE_THROWS_AS(this->train(knn_desc, homogen_table{}, homogen_table{}), domain_error);
+}
+
+KNN_BADARG_TEST("throws if train labels are empty") {
+    SKIP_IF(this->not_available_on_device());
+    const auto knn_desc = this->get_descriptor();
+    REQUIRE_THROWS_AS(this->train(knn_desc, this->get_train_data(), homogen_table{}), domain_error);
+}
+
+KNN_BADARG_TEST("throws if train data is empty") {
+    SKIP_IF(this->not_available_on_device());
+    const auto knn_desc = this->get_descriptor();
+    REQUIRE_THROWS_AS(this->train(knn_desc, homogen_table{}, this->get_train_labels()),
+                      domain_error);
+}
+
+KNN_BADARG_TEST(
+    "throws if the number of train samples is not equal to the number of train labels") {
+    SKIP_IF(this->not_available_on_device());
+    const auto knn_desc = this->get_descriptor();
+    REQUIRE_THROWS_AS(this->train(knn_desc, this->get_train_data(3), this->get_train_labels()),
+                      domain_error);
+    REQUIRE_THROWS_AS(this->train(knn_desc, this->get_train_data(), this->get_train_labels(3)),
+                      domain_error);
+}
+
+KNN_BADARG_TEST("throws if the number of columns in labels is greater then one") {
+    SKIP_IF(this->not_available_on_device());
+    const auto knn_desc = this->get_descriptor();
+    REQUIRE_THROWS_AS(this->train(knn_desc, this->get_train_data(3), this->get_train_labels(3, 2)),
+                      domain_error);
+}
+
+KNN_BADARG_TEST("accept if infer data has suitable shape and not zero") {
```

Comment:
What does "not zero" mean here?

Reply:
Fixed) Naming changed "not zero" -> "empty"